src/ol/render/canvas/canvasreplay.js

```diff
@@ -1440,13 +1484,20 @@ ol.render.canvas.PolygonReplay.prototype.setFillStrokeStyle = function(fillStyle
     var fillStyleColor = fillStyle.getColor();
     state.fillStyle = ol.colorlike.asColorLike(fillStyleColor ?
         fillStyleColor : ol.render.canvas.defaultFillStyle);
+    if (!this.transparency && ol.color.isRgba(state.fillStyle)) {
+      this.transparency = ol.color.asArray(
+          /** @type {ol.Color|string} */ (state.fillStyle))[0] != 1;
```

Comment:
Ok, I see that we won't get here with a pattern or gradient.  But I think things would be more efficient to have a `ol.color.isOpaque()` function.  And I think it is really only needed here to check strings.